google-cloud-storage/src/main/java/com/google/cloud/storage/UnifiedOpts.java

```diff
@@ -661,13 +669,209 @@ public Mapper<ListObjectsRequest.Builder> listObjects() {
+                BlobInfo.Builder::clearContentDisposition)
+            .put(BlobField.CONTENT_ENCODING.getGrpcName(), BlobInfo.Builder::clearContentEncoding)
+            .put(BlobField.CONTENT_LANGUAGE.getGrpcName(), BlobInfo.Builder::clearContentLanguage)
+            .put(BlobField.CONTENT_TYPE.getGrpcName(), BlobInfo.Builder::clearContentType)
+            .put(BlobField.CRC32C.getGrpcName(), BlobInfo.Builder::clearCrc32c)
+            .put(
+                BlobField.CUSTOMER_ENCRYPTION.getGrpcName(),
+                BlobInfo.Builder::clearCustomerEncryption)
+            .put(BlobField.CUSTOM_TIME.getGrpcName(), BlobInfo.Builder::clearCustomTime)
+            .put(BlobField.ETAG.getGrpcName(), BlobInfo.Builder::clearEtag)
+            .put(BlobField.EVENT_BASED_HOLD.getGrpcName(), BlobInfo.Builder::clearEventBasedHold)
+            .put(
+                BlobField.GENERATION.getGrpcName(),
+                b -> {
+                  BlobId current = b.getBlobId();
+                  return b.setBlobId(BlobId.of(current.getBucket(), current.getName()));
+                })
+            .put(BlobField.ID.getGrpcName(), BlobInfo.Builder::clearGeneratedId)
+            .put(BlobField.KMS_KEY_NAME.getGrpcName(), BlobInfo.Builder::clearKmsKeyName)
+            .put(BlobField.MD5HASH.getGrpcName(), BlobInfo.Builder::clearMd5)
+            .put(BlobField.MEDIA_LINK.getGrpcName(), BlobInfo.Builder::clearMediaLink)
+            .put(BlobField.METADATA.getGrpcName(), BlobInfo.Builder::clearMetadata)
+            .put(BlobField.METAGENERATION.getGrpcName(), BlobInfo.Builder::clearMetageneration)
+            .put(BlobField.OWNER.getGrpcName(), BlobInfo.Builder::clearOwner)
+            .put(
+                BlobField.RETENTION_EXPIRATION_TIME.getGrpcName(),
+                BlobInfo.Builder::clearRetentionExpirationTime)
+            .put(BlobField.SELF_LINK.getGrpcName(), BlobInfo.Builder::clearSelfLink)
+            .put(BlobField.SIZE.getGrpcName(), BlobInfo.Builder::clearSize)
+            .put(BlobField.STORAGE_CLASS.getGrpcName(), BlobInfo.Builder::clearStorageClass)
+            .put(BlobField.TEMPORARY_HOLD.getGrpcName(), BlobInfo.Builder::clearTemporaryHold)
+            .put(BlobField.TIME_CREATED.getGrpcName(), BlobInfo.Builder::clearCreateTime)
+            .put(BlobField.TIME_DELETED.getGrpcName(), BlobInfo.Builder::clearDeleteTime)
+            .put(
+                BlobField.TIME_STORAGE_CLASS_UPDATED.getGrpcName(),
+                BlobInfo.Builder::clearTimeStorageClassUpdated)
+            .put(BlobField.UPDATED.getGrpcName(), BlobInfo.Builder::clearUpdateTime)
+            .build();
+
+    private static final ImmutableMap<String, Mapper<BucketInfo.Builder>> bucketInfoFieldClearers =
+        ImmutableMap.<String, Mapper<BucketInfo.Builder>>builder()
+            .put(BucketField.ACL.getGrpcName(), BucketInfo.Builder::clearAcl)
+            // .put(BucketField.AUTOCLASS.getGrpcName(), b -> b.clearAutoclass())
```

Comment:
are we leaving this out because autoclass isn't out yet?

Reply:
Yeah, autoclass hasn't been wired into BucketInfo yet.